actionpack/lib/action_dispatch/middleware/templates/rescues/_message_with_did_you_mean.html.erb

```diff
@@ -0,0 +1,13 @@
+<% if defined?(DidYouMean) && DidYouMean.respond_to?(:correct_error) && exception.is_a?(DidYouMean::Correctable) %>
```

Comment:
I'm not excited about this.  Can we do something like `exception.respond_to?(:corrections)` or something?  There must be an easier way to detect if DYM is enabled.  Or we need to wrap this up in a method or something.

Reply:
The middle part seems off focus from the question we need to ask, but:

```suggestion
<% if defined?(DidYouMean::Correctable) && exception.is_a?(DidYouMean::Correctable) %>
```

.. seems reasonable to me: exceptions in particular can come from _anywhere_, so I'm hesitant to trust it to duck type as we expect just based on it having a particular method.

Reply:
I've changed it to @matthewd's suggestion, I can also change it to `exception.respond_to?(:corrections) && exception.respond_to?(:original_message)`.

Reply:
The conditional has been changed it to `exception.respond_to?(:original_message) && exception.respond_to?(:corrections)`.